cpp/rrtmgp/mo_gas_concentrations.h

```diff
@@ -102,6 +115,15 @@ class GasConcs {
     parallel_for( YAKL_AUTO_LABEL() , SimpleBounds<2>(nlay,ncol) , YAKL_LAMBDA (int ilay, int icol) {
       this_concs(icol,ilay,igas) = w;
     });
+#ifdef RRTMGP_ENABLE_KOKKOS
+    auto this_concs_k = this->concs_k;
+    Kokkos::parallel_for(nlay, KOKKOS_LAMBDA(int ilay) {
+      for (int icol = 0; icol < ncol; ++icol) {
```

Comment:
I'm curious why ncol is being serialized.

Reply:
@jgfouca adding to Matt's comment: if the view is LayoutLeft, then we can take a subview `subview(ALL,ALL,igas-1)`, and then just do `deep_copy(sv,w);`.

Reply:
Jim, you could also use MDRangePolicy, which I think it's the kokkos equivalent of SimpleBounds<N> in yakl.

Reply:
> @mrnorman , that's a good question. In YAKL, when you have something like this:
> 
> ```
> parallel_for( YAKL_AUTO_LABEL() , SimpleBounds<2>(nlay,ncol), YAKL_LAMBDA (int ilay, int icol) {
>    ...
> ```
> 
> Does that parallelize across both indices? I assumed it was only parallelizing on nlay. In Kokkos, parallelizing on both would require hierarchical parallelism which is a big step up in complexity since you need to set up thread teams for that.

YAKL parallelizes all dimensions added to the `Bounds` or `SimpleBounds` class in order from left to right being outermost to innermost in the threading / implied looping. More information here:

https://github.com/mrnorman/YAKL/wiki/parallel_for-and-Bounds-(C-and-Fortran-Styles),-and-synchronization

Reply:
Right, RangePolicy is roughly equivalent to a 1-D Bounds object, and MDRangePolicy is roughly equivalent to a multi-D Bounds object.

Reply:
Thanks guys, I will switch this pattern to MDRangePolicy

Reply:
Regarding your comment about hierarchical parallelism, YAKL can support multiple **tightly nested** loops by essentially collapsing them, which I assume is what `MDRangePolicy` does in its backend as well. YAKL does have two-level hierarchical parallelism, but it's not needed for this code.

Reply:
Thanks @mrnorman , that will simplify things a lot!